buildSrc/src/main/java/com/osacky/flank/gradle/SanityConfigValidation.kt

```diff
@@ -0,0 +1,45 @@
+package com.osacky.flank.gradle
+
+import org.gradle.api.GradleException
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+
+@Throws(GradleException::class)
+fun checkIfSanityAndValidateConfigs(config: FladleConfig) = when (config) {
+  is FlankGradleExtension -> config.checkAndValidateExtension()
+  is FladleConfigImpl -> config.checkAndValidateConfig()
+  else -> throw GradleException("Unable to check for sanity, check config type")
+}
+
+private fun FlankGradleExtension.checkAndValidateExtension() {
+  if (sanityRobo.getOrElse(false)) when {
+    instrumentationApk.isNotPresentOrBlank -> throwBaseConfigError("instrumentationApk")
+    additionalTestApks.isNotPresentOrEmpty -> throwBaseConfigError("additionalTestApks")
+    roboDirectives.isNotPresentOrEmpty -> throwBaseConfigError("roboDirectives")
+    roboScript.isNotPresentOrBlank -> throwBaseConfigError("roboScript")
+  }
+}
+
+private fun FladleConfigImpl.checkAndValidateConfig() {
+  if (sanityRobo.getOrElse(false)) when {
+    roboDirectives.isNotPresentOrEmpty -> throwAdditionalConfigError("roboDirectives", name)
```

Comment:
why is the ordering different for the property validation?
can we make both methods the same and just replace an empty base config name `""` with `base`?

Although `base` could be the name of the configuration. I don't have a better idea though.

Reply:
Sure, nice catch. That was old implementation before I've added some new properties to `FladleConfig`, thanks!